mdal/frmts/mdal_ascii_dat.cpp

```diff
@@ -444,6 +444,101 @@ void MDAL::DriverAsciiDat::readFaceTimestep(
   group->datasets.push_back( dataset );
 }
 
+bool MDAL::DriverAsciiDat::persist( MDAL::DatasetGroup *group )
+{
+  const bool isScalar = group->isScalar();
+  const bool isOnVerticies = group->isOnVertices();
+  std::string uri = group->uri();
+
+  if ( !MDAL::contains( uri, "_els_" ) && isOnVerticies == false )
+  {
+    // Should contain _els in name but it does not
+    uri.insert( uri.size() - 4, "_els_" );
+  }
+  else if ( !MDAL::contains( uri, "_vertex" ) && isOnVerticies == true )
+  {
+    // Should contain _vertex in name but it does not
+    uri.insert( uri.size() - 4, "_vertex" );
+  }
+
+  std::ofstream out( uri, std::ofstream::out );
+
+  // implementation based on information from:
+  // https://www.xmswiki.com/wiki/SMS:ASCII_Dataset_Files_*.dat
+  if ( !out )
+    return true; // Couldn't open the file
+
+  const Mesh *mesh = group->mesh();
+  size_t nodeCount = mesh->verticesCount();
+  size_t elemCount = mesh->facesCount();
+
+  out << "DATASET\n";
+  out << "OBJTYPE \"mesh2d\"\n";
+
+  if ( isScalar )
+    out << "BEGSCL\n";
+  else
+    out << "BEGVEC\n";
+
+  out << "ND " << nodeCount << "\n";
+  out << "NC " << elemCount << "\n";
+  out << "NAME " "\"" << group->name() << "\"" "\n";
+  std::string referenceTimeStr = group->referenceTime();
+
+  if ( !referenceTimeStr.empty() )
+  {
+    // Cutting of the JULIAN prefix
+    std::vector<std::string> referenceTimeStrWords = split( referenceTimeStr,  ' ' );
+
+    if ( referenceTimeStrWords.size() > 1 )
+      out << "RT_JULIAN " << referenceTimeStrWords[1] << "\n";
+    else
+      out << "RT_JULIAN " << referenceTimeStr << "\n";
+  }
+
+  out << "TIMEUNITS " << 0 << "\n";
+
+  for ( size_t time_index = 0; time_index < group->datasets.size(); ++ time_index )
+  {
+    const std::shared_ptr<MDAL::MemoryDataset> dataset
+      = std::dynamic_pointer_cast<MDAL::MemoryDataset>( group->datasets[time_index] );
+
+
+    out << "TS " << static_cast<bool>( dataset->active() ) << " " << std::to_string( dataset->time() ) << "\n";
```

Comment:
instead of `static_cast<bool>( dataset->active() )` it should be off for datasets on faces and of on datasets on vertices